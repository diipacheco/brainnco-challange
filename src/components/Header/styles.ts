import styled from 'styled-components';

export const Content = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;

  margin-top: 45px;

  @media screen and (min-width: 700px) {
    height: 100%;
    justify-content: space-around;
  }

  .logo-container {
    display: flex;
    flex-direction: column;

    justify-content: center;
    align-items: center;

    > img {
      height: 55px;
    }

    > h1 {
      color: var(--white);
      font-size: 26px;
      line-height: 37px;
      margin-top: 11px;
    }
  }

  p {
    margin-top: 88px;

    font-weight: 500;
    font-size: 14px;
    line-height: 17px;
    color: var(--white);
  }

  strong {
    display: block;
    font-size: 18px;
    line-height: 24px;

    color: #ffffff;
    margin-top: 06px;
  }
`;
