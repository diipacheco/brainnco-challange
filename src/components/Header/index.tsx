import React, { useEffect, useMemo } from 'react';

import { useLotteries } from '../../hooks/lotteries';
import useWindowSize from '../../hooks/useWindowSize';
import logo from '../../assets/logo.svg';

import Background from '../Background';
import Select from '../Select';

import { Content } from './styles';

const Header: React.FC = () => {
  const { selectedLottery, selectedContest, fetchSelectedContests, contest } =
    useLotteries();
  const { width } = useWindowSize();

  const formatContestDate = useMemo(() => {
    if (contest?.data) {
      const date = new Date(contest?.data);
      return Intl.DateTimeFormat('pt-br').format(date);
    }
  }, [contest]);

  useEffect(() => {
    fetchSelectedContests();
  }, [fetchSelectedContests]);
  return (
    <Background selectedLottery={selectedLottery.nome}>
      <Select />
      <Content>
        <div className="logo-container">
          <img src={logo} alt="logo brain challenge" />
          <h1>{selectedLottery.nome}</h1>
        </div>

        {width && width <= 700 ? (
          <p>{`CONCURSO Nº ${selectedContest?.concursoId}`}</p>
        ) : (
          <div>
            <p>Concurso</p>
            <strong>{`${selectedContest?.concursoId} – ${formatContestDate}`}</strong>
          </div>
        )}
      </Content>
    </Background>
  );
};

export default Header;
