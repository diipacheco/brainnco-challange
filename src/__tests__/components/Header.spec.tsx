import { render, act } from '@testing-library/react';

import Header from '../../components/Header';

jest.mock('../../hooks/lotteries', () => ({
  useLotteries: () => ({
    lotteries: [
      {
        id: 0,
        nome: 'mega-sena',
      },
      {
        id: 1,
        nome: 'quina',
      },
      {
        id: 2,
        nome: 'lotofácil',
      },
      {
        id: 3,
        nome: 'lotomania',
      },
      {
        id: 4,
        nome: 'timemania',
      },
      {
        id: 5,
        nome: 'dia de sorte',
      },
    ],
    fetchLotteries: jest.fn(),
    selectedLottery: {
      id: 0,
      nome: 'MEGA-SENA',
    },
    handleSelectedLottery: jest.fn(),
    fetchSelectedContests: jest.fn(),
    fetchContest: jest.fn(),
    contest: {
      data: '2021-08-25T04:32:48.857Z',
      id: '2359',
      loteria: 0,
      numeros: ['31', '32', '39', '42', '43', '51'],
    },
    selectedContest: {
      loteriaId: 0,
      concursoId: 2359,
    },
  }),
}));

describe('Header', () => {
  it('should match snapshot', () => {
    const wrapper = render(<Header />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should be able to render the lottery name', () => {
    const { container } = render(<Header />);

    const lotteryNameElement = container.querySelector('h1');

    expect(lotteryNameElement?.innerHTML).toBe('MEGA-SENA');
  });

  it('should be able to render the lottery contest number and date', () => {
    const { container } = render(<Header />);

    const lotteryContestNumberAndDateElement =
      container.querySelector('strong');

    expect(lotteryContestNumberAndDateElement?.innerHTML).toBe(
      '2359 – 25/08/2021',
    );
  });

  it('should render only the contest number on mobile devices', () => {
    act(() => {
      window = Object.assign(window, { innerWidth: 700 });
    });

    const { getByText } = render(<Header />);

    const lotteryContestNumberElement = getByText('CONCURSO Nº 2359');

    expect(lotteryContestNumberElement).toBeInTheDocument();
  });
});
