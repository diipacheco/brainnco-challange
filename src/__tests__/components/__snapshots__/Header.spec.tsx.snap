// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  width: 100%;
  height: 390px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  background-color: var(--green-light);
}

.c1 {
  width: 100%;
  max-width: 215px;
  padding: 17px 33px;
  font-weight: 500;
  font-size: 15px;
  line-height: 18px;
  border-radius: 05px;
  background-color: var(--white);
  margin-top: 30px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-top: 45px;
}

.c2 .logo-container {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 .logo-container > img {
  height: 55px;
}

.c2 .logo-container > h1 {
  color: var(--white);
  font-size: 26px;
  line-height: 37px;
  margin-top: 11px;
}

.c2 p {
  margin-top: 88px;
  font-weight: 500;
  font-size: 14px;
  line-height: 17px;
  color: var(--white);
}

.c2 strong {
  display: block;
  font-size: 18px;
  line-height: 24px;
  color: #ffffff;
  margin-top: 06px;
}

@media screen and (min-width:700px) {
  .c0 {
    height: 100vh;
    max-width: 550px;
  }

  .c0::after {
    content: '';
    width: 40px;
    height: 100%;
    background-color: var(--background);
    position: absolute;
    border-radius: 62%;
    right: -19px;
  }
}

@media screen and (max-width:700px) {
  .c0::after {
    content: '';
    width: 100%;
    height: 52px;
    background-color: var(--background);
    position: absolute;
    border-radius: 40%;
    top: 92%;
  }
}

@media screen and (min-width:700px) {
  .c2 {
    height: 100%;
    -webkit-box-pack: space-around;
    -webkit-justify-content: space-around;
    -ms-flex-pack: space-around;
    justify-content: space-around;
  }
}

<body>
    <div>
      <div
        class="c0"
      >
        <select
          class="c1"
          data-testid="select"
        >
          <option
            data-testid="selected-option-0"
            value="mega-sena"
          >
            MEGA-SENA
          </option>
          <option
            data-testid="selected-option-1"
            value="quina"
          >
            QUINA
          </option>
          <option
            data-testid="selected-option-2"
            value="lotofácil"
          >
            LOTOFÁCIL
          </option>
          <option
            data-testid="selected-option-3"
            value="lotomania"
          >
            LOTOMANIA
          </option>
          <option
            data-testid="selected-option-4"
            value="timemania"
          >
            TIMEMANIA
          </option>
          <option
            data-testid="selected-option-5"
            value="dia de sorte"
          >
            DIA DE SORTE
          </option>
        </select>
        <div
          class="c2"
        >
          <div
            class="logo-container"
          >
            <img
              alt="logo brain challenge"
              src="logo.svg"
            />
            <h1>
              MEGA-SENA
            </h1>
          </div>
          <div>
            <p>
              Concurso
            </p>
            <strong>
              2359 – 25/08/2021
            </strong>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdnxRM hxsYFM"
    >
      <select
        class="sc-gtsrHT fUpVAm"
        data-testid="select"
      >
        <option
          data-testid="selected-option-0"
          value="mega-sena"
        >
          MEGA-SENA
        </option>
        <option
          data-testid="selected-option-1"
          value="quina"
        >
          QUINA
        </option>
        <option
          data-testid="selected-option-2"
          value="lotofácil"
        >
          LOTOFÁCIL
        </option>
        <option
          data-testid="selected-option-3"
          value="lotomania"
        >
          LOTOMANIA
        </option>
        <option
          data-testid="selected-option-4"
          value="timemania"
        >
          TIMEMANIA
        </option>
        <option
          data-testid="selected-option-5"
          value="dia de sorte"
        >
          DIA DE SORTE
        </option>
      </select>
      <div
        class="sc-dlnjwi dTFYgY"
      >
        <div
          class="logo-container"
        >
          <img
            alt="logo brain challenge"
            src="logo.svg"
          />
          <h1>
            MEGA-SENA
          </h1>
        </div>
        <div>
          <p>
            Concurso
          </p>
          <strong>
            2359 – 25/08/2021
          </strong>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
